src/cmap/connection_pool.ts

```diff
@@ -289,9 +345,19 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
    * Pool reset is handled by incrementing the pool's generation count. Any existing connection of a
    * previous generation will eventually be pruned during subsequent checkouts.
    */
-  clear(): void {
-    this[kGeneration] += 1;
-    this.emit('connectionPoolCleared', new ConnectionPoolClearedEvent(this));
+  clear(serviceId?: ObjectId): void {
+    if (this.loadBalanced && serviceId) {
+      const generation = this.serviceGenerations.get(serviceId);
```

Comment:
what if `!has(serviceId)`?

Reply:
Based on other update an exception will now be thrown in this case if the `serviceId` is missing.